packages/app-page-builder/src/pageEditor/plugins/elementSettings/UnlinkBlockAction.ts

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback } from "react";
+import { useEventActionHandler } from "~/editor/hooks/useEventActionHandler";
+import { UpdateElementActionEvent } from "~/editor/recoil/actions";
+import { activeElementAtom, elementByIdSelector } from "~/editor/recoil/modules";
+import { useRecoilValue } from "recoil";
+
+interface UnlinkBlockActionPropsType {
+    children: React.ReactElement;
+}
+const UnlinkBlockAction: React.FC<UnlinkBlockActionPropsType> = ({ children }) => {
+    const eventActionHandler = useEventActionHandler();
+    const activeElementId = useRecoilValue(activeElementAtom);
+    const element = useRecoilValue(elementByIdSelector(activeElementId as string));
+
+    const onClick = useCallback((): void => {
+        if (element) {
+            eventActionHandler.trigger(
+                new UpdateElementActionEvent({
+                    element: {
+                        ...element,
+                        data: newData
+                    },
+                    history: true
+                })
+            );
+        }
+    }, [activeElementId]);
```

Comment:
This whole section can be simplified; I added lots of hooks for this in https://github.com/webiny/webiny-js/tree/page-builder-blocks-v2-disable-block-element-updated/packages/app-page-builder/src/editor/hooks.

What you need are 2 hooks: `useActiveElement` and `useUpdateElement`, and this whole mess will go away.